### PR TITLE
Add API doc

### PR DESCRIPTION
**Pull request by bac**
API doc converted from Google docs, stripped of discussion, and formatted as mark down.

Some light editing was done when the text seemed obviously to be in error.  Those are marked like ==changed==.  Any section with == formatting is a comment to the reviewer to check for accuracy against the original document.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/680/
Test PASSed.

File: docs/API.md [hunk 1]
```diff
@@ -0,0 +1,1611 @@
...
+}
+```
+
+Example:
+
+`GET ubuntu/meta/any?include=color&include=archive-size&include=extra-info/featured`
+
+```
+{
+                "Id": "trusty/ubuntu-3",
```
Comment (mitechie) [inline docs/API.md]: the indention seems a bit off here

File: docs/API.md [hunk 2]
```diff
@@ -0,0 +1,1611 @@
...
+```
+
+
+Example:
+
+`GET mediawiki/meta/bundle-metadata`
+
+```
+{
+                "Services": {
```
Comment (mitechie) [inline docs/API.md]: same here

**Comment by mitechie** [comment]
ty, as a first pass this is great. There's some indention cleanup that would be great to get into this first update and we can come back later and look if there's ways to do a ToC at the top and such.

:+1:

**Comment by mitechie** [comment]
/me pretends to be @hatched 

```
it would be awesome if this had some header links
would also be nice if the api calls had working copy/paste examples
looks good beyond those couple things - I'm sure overtime we can tweak the code examples so they are all formatted the same
```

:+1:

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/681/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**